### PR TITLE
feat: vimium compat (not complete)

### LINKED ISSUE #478: [Demand] Allow users to disable Advanced Security Mode
## Metadata

Field | Value
--- | ---
Figma Link | -
Expected UX prototype due | -
Actual UX prototype release | -
Expected engineering due | -
Corresponding Pull Request | -
Completion Version | -

## Background

Some users may experience incompatibility with certain applications which also use Shadow Root. We will offer them a way to disabled Advanced Security Mode in Maskbook, so that they may continue using both.

## Content

We will add a switch option in Settings panel, in Advanced 

- Title:
  - en: Advanced Security Mode
  - zh: 高級安全模式
- Tooltip:
  - en: Display confidential information in Shadow Root
  - zh: 敏感信息显示于 Shadow Root 内部
- Right Component:
  - Switch (Default: True)

## Notes

- Unimportant feature, do when free.
- Dependencies:
  - #369
  - #479

### PR DESCRIPTION
**Pull request by Jack-Works**
<del>not going to close</del> #478

### CODE REVIEW

File: src/_locales/zh/messages.json [hunk 1]
```diff
@@ -192,6 +192,8 @@
     "set_as_default": "設為默認",
     "set_up": "開始",
     "settings": "設定",
+    "settings_advance_security": "高級安全模式",
+    "settings_advance_security_desc": "將内容展示在影子根中，以免於社交網路之窺探。",
```
Comment (guanbinrui) [inline src/_locales/zh/messages.json]: No `settings_vim_compat` here?

File: src/components/shared-settings/settings.ts [hunk 2]
```diff
@@ -30,6 +30,18 @@ export const disableOpenNewTabInBackgroundSettings = createNewSettings<boolean>(
     },
 )
 
+export const renderInShadowRootSettings = createNewSettings<boolean>(
+    'render in shadow root',
+    /**
+     * ? In WKWebview, the web extension polyfill is not ready for it.
+     */
+    webpackEnv.target === 'WKWebview' ? false : true,
```
Comment (guanbinrui) [inline src/components/shared-settings/settings.ts]: `webpackEnv.target !== 'WKWebview'`

File: src/_locales/zh/messages.json (hunk 1 above)
Comment (Jack-Works) [reply src/_locales/zh/messages.json]: following the suggestion from #478 @neruthes

File: src/components/shared-settings/settings.ts (hunk 2 above)
Comment (Jack-Works) [reply src/components/shared-settings/settings.ts]: it named `renderInShadowRootSettings` internally. So `true` will use ShadowRoot. `false` won't

File: src/_locales/zh/messages.json (hunk 1 above)
Comment (guanbinrui) [reply src/_locales/zh/messages.json]: I mean `settings_vim_compat` exists in the English version but absents here.

File: src/components/shared-settings/settings.ts (hunk 2 above)
Comment (guanbinrui) [reply src/components/shared-settings/settings.ts]: Yeah. The statement can be shorter.

File: src/utils/jss/renderInShadowRoot.tsx [hunk 3]
```diff
@@ -3,32 +3,66 @@ import { jssPreset, StylesProvider, ThemeProvider } from '@material-ui/styles'
...
+                        </RenderInShadowRootWrapper>
+                    </ErrorBoundary>,
+                    element,
+                )
                 rendered = true
             }
         })
+    }
+    // ? Wait for the renderInShadowRootSettings.value to be sync
+    setTimeout(() => tryRender, 400)
```
Comment (guanbinrui) [inline src/utils/jss/renderInShadowRoot.tsx]: The `tryRender` never runs. Do you mean `setTimeout(tryRender, 400)`?

**Pull request merged**